src/cli/pubkey.cpp

```diff
@@ -215,6 +217,15 @@ class PK_Sign final : public Command
          this->read_file(get_arg("file"), onData);
 
          output() << Botan::base64_encode(signer.signature(rng())) << "\n";
+
+         if(key->stateful_operation())
+            {
+            std::ofstream updated_key(key_file);
+            if(passphrase.empty())
+               updated_key << Botan::PKCS8::PEM_encode(*key);
+            else
+               updated_key << Botan::PKCS8::PEM_encode(*key, rng(), passphrase);
+            }
          }
```

Comment:
The signature output should be generated here instead of in [L219](https://github.com/randombit/botan/blob/db1b06d2ee44a1bc72de3bdd4c4098cc8c149dcc/src/cli/pubkey.cpp#L219). This has been corrected in #2069.